models/Image.js

```diff
@@ -1,76 +1,160 @@
+const {PubSub} = require('@google-cloud/pubsub');
+const errors = require('@twreporter/errors').default;
 var config = require('../config');
 var extractIPTC = require('../lib/extractIPTC');
 var keystone = require('@twreporter/keystone');
-var resizeImage = require('../lib/resizeImage');
 var transform = require('model-transform');
 var Types = keystone.Field.Types;
 
 var Image = new keystone.List('Image', {
-    map: { name: 'description' }
+  map: { name: 'description' }
 });
-var bucket = config['options']['gcs config']['bucket']
+
+const bucket = config['options']['gcs config']['bucket'];
+
+const resizedOpts = [['desktop', 2000], ['tablet', 1200], ['mobile', 800], ['w400', 400], ['tiny', 150]];
+
+const outputBucket = process.env.KEYSTONE_PUBSUB_RESIZE_IMAGE_OUTPUT_BUCKET || 'twreporter-multimedia';
+const topicName = process.env.KEYSTONE_PUBSUB_RESIZE_IMAGE_TOPIC || 'resize-image';
+const resizeImageURLPrefix = process.env.KEYSTONE_RESIZE_IMAGE_URL_PREFIX || 'https://www.twreporter.org/images/'
+
+/**
+ *  Publish resize events through Google Cloud Pub/Sub.
+ *
+ *  @param {string} filename - uploaded file name
+ *  @returns {Promise} resolve with an array of message ids or reject with error
+ */
+function publishResizeTasks(filename) {
+  const inputBucket = bucket;
+  const pubSubClient = new PubSub()
+
+  const promises = []
+
+  resizedOpts.forEach((resizedOpt) => {
+    let [resolution, resizeWidth] = resizedOpt;
+    const promise = pubSubClient
+      .topic(topicName)
+      .publish(Buffer.from(filename), {
```

Comment:
could we have the following attributes here?

```
{
  "input_bucket": "...",
  "output_bucket": "...",
  "target": "...",
  "resize_width": "...",
  "original_width": "...",
  "description": "..."    
}
```

The reason to require `original_width` is that we can save the duplicated decode effort on cloud function to only retrieve the original width.
For `description` field, it is suggested by photographers to identify the image.